include/revng/Pipeline/Context.h

```diff
@@ -120,11 +123,47 @@ class Context {
 
 public:
   llvm::Error storeToDisk(llvm::StringRef Path) const {
-    return Globals.storeToDisk(Path);
+    if (auto Error = Globals.storeToDisk(Path); Error)
+      return Error;
+
+    return llvm::Error::success();
   }
+
   llvm::Error loadFromDisk(llvm::StringRef Path) {
-    return Globals.loadFromDisk(Path);
+    if (auto Error = Globals.loadFromDisk(*this, Path); Error)
+      return Error;
+
+    return llvm::Error::success();
+  }
+
+public:
+  void getTargetsAssociatedToPaths(const TupleTreePath &Path,
+                                   InvalidationMap &Out) const {
+    for (auto &Name : Globals.getGlobalNames()) {
+      llvm::cantFail(Globals.get(Name))->getTargetsAssociatedToPaths(Path, Out);
+    }
   }
-};
 
+  bool isTargetCached(const LocatedTarget &Target) const {
+    for (auto &Name : Globals.getGlobalNames()) {
+      if (llvm::cantFail(Globals.get(Name))->isTargetCached(Target))
+        return true;
+    }
+    return false;
+  }
+  void collectReadFields(const LocatedTarget &Target) const {
+    Globals.collectReadFields(Target);
+  }
+
+  void clearReadFields() const { Globals.clearReadFields(); }
+  void pushReadFields() const { Globals.pushReadFields(); }
+  void popReadFields() const { Globals.popReadFields(); }
+  void stopTracking() const { Globals.stopTracking(); }
+
+  Committer makeCommitter() { return Committer(*this, CurrentStepName); }
```

Comment:
The whole `Committer` infrastructure is undocumented but it's very important for devs to understand it.

Please thoroughly document it, possibly with examples.